ui/imports/shared/popups/send/models/SendModalAssetsAdaptor.qml

```diff
@@ -0,0 +1,209 @@
+            readonly property bool marketDetailsLoading: model.detailsLoading
+            readonly property real marketPrice: marketDetails.currencyPrice.amount ?? 0
+            readonly property real marketChangePct24hour: marketDetails.changePct24hour ?? 0
+
+            readonly property bool visible: {
+                root.controller.revision
+
+                if (!root.controller.filterAcceptsSymbol(model.symbol))
+                    return false
+
+                if (isCommunityAsset) {
+                    return root.showCommunityAssets
+                }
+                return currentCurrencyBalance >= root.marketValueThreshold
+            }
+
+            readonly property url icon: !!model.image ? model.image : Constants.tokenIcon(model.symbol, false)
+
+            readonly property var balancesModel: filteredBalances
+
+            SortFilterProxyModel {
+                id: filteredBalances
+
+                sourceModel: rootModel.balances
+
+                filters: [
+                    FastExpressionFilter {
+                        expression: root.account === model.account
+                        expectedRoles: ["account"]
+                    }
+                ]
+            }
+
+            FunctionAggregator {
+                id: totalBalanceAggregator
+
+                model: filteredBalances
+                initialValue: "0"
+                roleName: "balance"
+
+                aggregateFunction: (aggr, value) => AmountsArithmetic.sum(
+                                       AmountsArithmetic.fromString(aggr),
+                                       AmountsArithmetic.fromString(value)).toString()
+            }
+
+            FunctionAggregator {
+                id: chainsAggregator
+
+                readonly property var uniqueChains: [...new Set(value).values()]
+
+                model: filteredBalances
+                initialValue: []
+                roleName: "chainId"
+
+                aggregateFunction: (aggr, value) => [...aggr, value]
+            }
+        }
+
+        expectedRoles:
+            ["tokensKey", "symbol", "image", "balances", "decimals",
+             "detailsLoading", "marketDetails", "communityId", "addressPerChain"]
+        exposedRoles:
+            ["key", "error", "currentBalance", "currentCurrencyBalance", "currentBalanceText",
+             "icon", "visible", "marketDetailsAvailable", "marketDetailsLoading",
+             "marketPrice", "marketChangePct24hour", "isCommunityAsset", "balancesModel"]
+
+
+        /* Internal function to search token address */
+        function __searchAddressInList(addressPerChain, searchString) {
+            let addressFound = false
+            let tokenAddresses = ModelUtils.modelToFlatArray(addressPerChain, "address")
+            for (let i =0; i< tokenAddresses.length; i++){
+                if(tokenAddresses[i].toUpperCase().startsWith(searchString.toUpperCase())) {
```

Comment:
Could be optimized... why call `searchString.toUpperCase()` again and again for every `for` iteration?